content/extensions/ui-extensions.md

```diff
@@ -0,0 +1,468 @@
+
+### Using Curl
+
+#### 1. Retrieve your credentials from your targeted PIM
+
+::: info
+🛠 For every call to the API, you will need `X-PIM-TOKEN` & `X-PIM-CLIENT-ID`.
+
+In this example, we will create a new `connection` in the PIM and use it to generate an `API token`.
+:::
+
+**1. Create a connection in Akeneo PIM:**
+- Navigate to **Connect** > **Connection settings** > **Create**.
+- Fill out the form to create the connection.
+- Note the generated `Client ID`, `Secret`, `Username`, and `Password`.
+
+**2. Set Your Environment Variables:**
+- Define the Client ID, Secret, Username, Password, and Akeneo host URL as environment variables:
+
+   ```bash [snippet:Shell]
+        export CLIENT_ID="your-client-id"
+    export CLIENT_SECRET="your-client-secret"
+    export API_USERNAME="your-API-username"
+    export API_PASSWORD="your-API-password"
+    export TARGET_PIM_URL="https://your-pim.cloud.akeneo.com"
+   ```
+Replace the placeholders with your actual credentials and host URL.
+
+**3. Encode Your Credentials:**
+- Encode the Client ID and Secret in base64 format, separated by a colon `:`:
+   ```bash [snippet:Shell]
+        export BASE64_ENCODED_CLIENTID_AND_SECRET=$(echo -n "$CLIENT_ID:$CLIENT_SECRET" | base64 -w 0)
+   // For Mac OS user remove the -w 0 option
+   ```
+
+**4.  Your API Token:**
+- Make the API call to retrieve your `API token` using the environment variables:
+   ```bash [snippet:Shell]
+        curl --request POST "$TARGET_PIM_URL/api/oauth/v1/token" \
+    --header "Content-Type: application/json" \
+    --header "Authorization: Basic $BASE64_ENCODED_CLIENTID_AND_SECRET" \
+    --data-raw '{
+    "grant_type": "password",
+    "username": "'"$API_USERNAME"'",
+    "password": "'"$API_PASSWORD"'"
+    }'
+   ```
+After retrieving the API token, store the `access_token` from the response in an environment variable:
+   ```bash [snippet:Shell]
+        export PIM_API_TOKEN="..."
+    // Replace with the actual token from the response
+   ```
+
+::: info
+ℹ️ Note that the token has a lifespan of one hour.
+:::
+
+::: info
+🛠 Custom apps are also supported. To use one, add a variable `app_access_token` with your API Token.
+:::
+
+#### 2. Create a UI extension
+You can create a UI extension once you have a valid PIM API token.
+
+```bash [snippet:Shell]
+    curl --request POST "$TARGET_PIM_URL/api/rest/v1/ui-extensions" \
+--header "Authorization: Bearer $PIM_API_TOKEN" \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "name": "my_awesome_extension",
+    "version": "V1.02",
+    "type": "link",
+    "position": "pim.product.header",
+    "configuration": {
+        "url": "https://myapp.com/product",
+        "default_label": "My awesome extension",
+        "labels": {
+          "en_US": "My awesome extension"
+        }
+    }
+}'
+```
+
+#### 3. Update a UI extension
+To update a UI extension, you must possess a valid PIM API token and the UUID of the extension you want to update.
```

Comment:
```suggestion
To update a UI extension, you must have a valid PIM API token and the UUID of the extension you want to update.
```

Possess is used for real possession (I possess a computer) or when a demon has complete power on you ;)

"have" is enough